### PR TITLE
[simx vt] Add Mellanox SimX virtual testbed infra 

### PR DESCRIPTION
**Pull request by mykolaf**
### Description of PR

Summary:
Fixes # (issue)

Add a few playbooks which allow to use t0 virtual testbed with Mellanox SimX simulator.
SimX is a functional simulation of Mellanox PCI devices
 - It simulates 
   - Mellanox NICs
   - Switch: Spectrum-1, Spectrum-2, Spectrum-3
- It simulates both FW and HW
- All flows are supported (controll + data path)

Mellanox only provides the simulator to it's customers.
Without the simulator none of this scripts are relevant.

### Type of change

- [] Bug fix
- [x] Testbed and Framework(new/improvement)
- [] Test case(new/improvement)

#### How did you do it?
Add start/stop playbooks for new DUT type=simx.
The playbook starts the Simulator inside a docker container(for ease of deployment) and then wm-topology module can connect it to the topology.
#### How did you verify/test it?
```
$ ./testbed-cli.sh -m veos.vtb start-vms server_1 password.txt
$ ./testbed-cli.sh -t vtestbed.csv -m veos.vtb add-topo vms-kvm-t0 password.txt
```
- run BGP test

### CODE REVIEW

File: ansible/roles/vm_set/tasks/start_sid.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    dest: "{{ home_path }}/sid/images/settings.ini"
+
+- name: start SimX in docker container
+  shell: /usr/bin/env python {{ home_path }}/sid/images/start.py
+
+- name: Wait for SID to start
+  pause:
+    seconds: 60
+
+# Annoying linux bridges will be gone in later SimX versions
```
Comment (liat-grozovik) [inline ansible/roles/vm_set/tasks/start_sid.yml]: remove

File: ansible/veos.vtb [hunk 2]
```diff
@@ -31,3 +31,4 @@ topologies=['t1', 't1-lag', 't1-64-lag', 't0', 't0-16', 't0-56', 't0-52', 'ptf32
 [sonic]
 vlab-01 ansible_host=10.250.0.101 type=kvm hwsku=Force10-S6000
 vlab-02 ansible_host=10.250.0.102 type=kvm hwsku=Force10-S6100
+vlab-simx-01 ansible_host=10.250.0.103 type=simx hwsku=MSN2700
```
Comment (liat-grozovik) [inline ansible/veos.vtb]: what about spectrum-2 based systems?

File: ansible/doc/README.testbed.SimXSetup.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+Instructions to build or download pre-built docker-ptf [here](https://github.com/Azure/sonic-mgmt/blob/master/ansible/doc/README.testbed.VsSetup.md#setup-docker-registry-for-ptf-docker)
+
+## Build or download *sonic-mgmt* docker image
+
+Instructions to set up sonic-mgmt docker [here](https://github.com/Azure/sonic-mgmt/blob/master/ansible/doc/README.testbed.VsSetup.md#build-or-download-sonic-mgmt-docker-image)
+
+## Download SimX in Docker  
+
+- Download SiD image from [here](https://mellanox365.sharepoint.com/:u:/s/sw/Systems/ETME0APOkf9DsyKneVwXpz8BuXZhgdluxv76WAULxlD96g?e=OU48ja)
```
[the remark this answers is not in the document]
Comment (mykolaf) [reply ansible/doc/README.testbed.SimXSetup.md]: Removed

File: ansible/doc/README.testbed.SimXSetup.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+root      1144  160 21.7 24598832 7151752 ?    Sl   07:45 464:30 /opt/simx/bin/simx-qemu-system-x86_64
+```
+
+## Deploy minigraph on the DUT
+
+```
+$ ./testbed-cli.sh -t vtestbed.csv -m veos.vtb deploy-mg vms-kvm-t0 lab password.txt
+```
+
+You should be login into the SImX vm using IP: 10.250.0.103 using admin:YourPaSsWoRd.
```
[the remark this answers is not in the document]
Comment (mykolaf) [reply ansible/doc/README.testbed.SimXSetup.md]: Fixed

File: ansible/doc/README.testbed.SimXSetup.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+There is also a telnet connection: 
+
+```
+$ telnet localhost 1213
+```
+
+You should see BGP sessions up in sonic.
```
[the remark this answers is not in the document]
Comment (mykolaf) [reply ansible/doc/README.testbed.SimXSetup.md]: Done

File: ansible/roles/vm_set/library/simx_port.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/python
```
[the remark this answers is not in the document]
Comment (mykolaf) [reply ansible/roles/vm_set/library/simx_port.py]: Done

File: ansible/roles/vm_set/tasks/start_sid.yml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  file: path={{ item }} state=directory mode=0755
+  with_items:
+    - "sid/images"
+    - "sid/disks"
+
+- include_vars: group_vars/sonic/vars
+
+# TODO: handle serial port
+- set_fact:
+    src_disk_image: "{{ home_path }}/sid/images/sonic_spc.img"
```
[the remark this answers is not in the document]
Comment (mykolaf) [reply ansible/roles/vm_set/tasks/start_sid.yml]: Now will use sonic.img for any spectrum-x based system.

File: ansible/veos.vtb (hunk 2 above)
Comment (mykolaf) [reply ansible/veos.vtb]: Added entry for MSN3700

**Pull request merged**